### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by emord**
https://docs.google.com/spreadsheets/d/1FCBPllydWQiwDJ3d8wWLlquRKVUIQytCnR_sTMEHQvQ/edit#gid=0

elf: @czue

### CODE REVIEW

File: corehq/apps/app_manager/decorators.py [hunk 1]
```diff
@@ -51,4 +51,4 @@ def _no_conflict(*args, **kwargs):
     return _no_conflict
 
 require_can_edit_apps = require_permission(Permissions.edit_apps)
-require_deploy_apps = login_and_domain_required  # todo: can fix this when it is better supported
+require_deploy_apps = require_can_edit_apps
```
Comment (czue) [inline corehq/apps/app_manager/decorators.py]: I'm not 100% sure this is right. There are people who will want to install apps on phones but should not be able to edit apps. How is this used exactly?

File: corehq/apps/dashboard/views.py [hunk 2]
```diff
@@ -114,8 +114,7 @@ def check_permissions(self, in_data):
 def _get_default_tile_configurations():
     can_edit_data = lambda request: (request.couch_user.can_edit_data()
                                      or request.couch_user.can_export_data())
-    can_edit_apps = lambda request: (request.couch_user.is_web_user()
-                                     or request.couch_user.can_edit_apps())
+    can_edit_apps = lambda request: (request.couch_user.can_edit_apps())
```
Comment (czue) [inline corehq/apps/dashboard/views.py]: likewise guessing this was intentional due to the above situation

File: corehq/apps/hqmedia/views.py [hunk 3]
```diff
@@ -490,6 +490,7 @@ def log_errors(self, errors):
         )
 
     @method_decorator(safe_download)
+    @method_decorator(require_permission(Permissions.edit_apps, login_decorator=login_with_permission_from_post()))
```
Comment (czue) [inline corehq/apps/hqmedia/views.py]: does the phone never hit this URL? that's one situation we have to be careful of.

File: corehq/apps/app_manager/decorators.py (hunk 1 above)
Comment (czue) [inline corehq/apps/app_manager/decorators.py]: I think the "right" behavior would be to make a new permission for being able to deploy apps that encompasses some of this, but not sure how hard that is. our permissions framework is not all that extensible which is why it's so limited.

**Comment by czue** [comment]
@amsagoff do you have any memory of conversations related to read only users needing to access various parts of app installation?

**Comment by czue** [comment]
put a don't merge label on this. @emord can you take a look at my comments and see what you think?

**Comment by czue** [comment]
@amsagoff bump on previous question

**Comment by emord** [comment]
@czue I started looking more deeply into it, and after your comments I think I should look more deeply and do it in parts. I'm going to close this for now and do it piecewise to make sure users won't have any weird side effects.

**Comment by czue** [comment]
roger

**Comment by amsagoff** [comment]
Hey,

I haven't been completely following this convo, but is it right that we're
not clear on the spec? Thinking of working with Kriti to get a more solid,
signed off spec down and hten we can implement. LMK if i'm reading the
blocker right.

**Comment by czue** [comment]
that sounds good. the main question we were discussing on this PR was whether read only users should be able to "deploy" apps, which I remember coming up before and thought we decided they should.

**Comment by amsagoff** [comment]
ah yes. i'd say make new version, no. but deploy, yes.

I'll work with kriti on getting this all down

**Pull request closed without merging**